File: sicp/1-building-abstractions-with-procedures/1-1/1-1-4/1.php
<?php

function square($x)
{
    return $x * $x;
}

echo square(5);

function sum_of_square($x, $y)
{
    return square($x) + square($y);
}

echo sum_of_square(2, 3);

$a = function($x) { return $x * $x; };

echo $a(5); //25

File: sicp/1-building-abstractions-with-procedures/1-1/1-1-4/1.rs
fn square(x: i32) -> i32 {
    return x * x;
}

fn sum_of_square(x: i32, y: i32) -> i32 {
    return square(x) + square(y);
}

fn main() {
    println!("{}", square(5));
    println!("{}", sum_of_square(2, 3));

    let a = |x: i32| x * x;
    println!("{}", a(5));
}
